src/components/Dashboard/RewardsModule.js

```diff
@@ -0,0 +1,343 @@
+import React from 'react'
+import {
+  Box,
+  Button,
+  GU,
+  Help,
+  Info,
+  textStyle,
+  useTheme,
+  Link,
+} from '@aragon/ui'
+
+import Loading from './Loading'
+import NoRewards from './NoRewards'
+
+import { useCourtConfig } from '../../providers/CourtConfig'
+import { useConnectedAccount } from '../../providers/Web3'
+
+import { formatTokenAmount, bigNum } from '../../lib/math-utils'
+
+const useTotalDisputesFees = (arbitrableFees, appealFees) => {
+  const totalArbitrableFees = arbitrableFees
+    ? arbitrableFees.reduce((acc, fee) => acc.add(fee.amount), bigNum(0))
+    : bigNum(0)
+
+  const totalAppealFees = appealFees
+    ? appealFees.reduce((acc, fee) => acc.add(fee.amount), bigNum(0))
+    : bigNum(0)
+
+  return { totalArbitrableFees, totalAppealFees }
+}
+
+// Ruling Fees => ANJ fees
+// Dispute fees + Appeal Fees => DAI fees
+const RewardsModule = React.memo(function RewardsModule({
+  rewards,
+  loading,
+  onSettleReward,
+  onSettleAppealDeposit,
+}) {
+  const connectedAccount = useConnectedAccount()
+
+  // Form submission
+  const handleFormSubmit = async event => {
+    event.preventDefault()
+
+    if (!rewards) return
+
+    try {
+      // Claim all arbitrable fee rewards
+      for (const arbitrableFee of rewards.arbitrableFees) {
+        const { disputeId, rounds } = arbitrableFee
+        for (const roundId of rounds) {
+          const rewardTx = await onSettleReward(
+            disputeId,
+            roundId,
+            connectedAccount
+          )
+          await rewardTx.wait()
+        }
+      }
+
+      // Claim all appeal fee rewards
+      for (const appealFee of rewards.appealFees) {
+        const { disputeId, rounds } = appealFee
+        for (const roundId of rounds) {
+          const rewardTx = await onSettleAppealDeposit(disputeId, roundId)
+          await rewardTx.wait()
+        }
+      }
+    } catch (err) {
+      console.log(`Error claiming rewards: ${err}`)
+    }
+  }
+
+  const hasRewardsToClaim =
+    rewards &&
+    (rewards.rulingFees.gt(0) || rewards.totalDisputesFees.length > 0)
+
+  const showHeading = !loading && hasRewardsToClaim
+
+  return (
+    <Box
+      heading={showHeading && 'Rewards'}
+      padding={hasRewardsToClaim ? 0 : 3 * GU}
+    >
+      {(() => {
+        if (loading) return <Loading height={150} />
+
+        if (!hasRewardsToClaim) return <NoRewards />
```

Comment:
Just to be explicit :)

```suggestion
        if (!hasRewardsToClaim) return <NoRewards />
        return null
```